lib/plugins/aws/package/compile/events/schedule.js

```diff
@@ -113,18 +128,38 @@ class AwsCompileScheduledEvents {
                 );
               }
 
+              if (method === 'scheduler' && !roleArn) {
```

Comment:
This is the execution role that the Scheduler will use to invoke the target lambda? How would you feel about generating a default execution role if a `roleArn` was not provided?

As this PR currently stands, a minimal execution role would look like the following:

```json
"ScheduleExecutionRole": {
  "Type": "AWS::IAM::Role",
  "Properties": {
    "AssumeRolePolicyDocument": {
      "Version": "2012-10-17",
      "Statement": [
        {
          "Effect": "Allow",
          "Principal": {
            "Service": "scheduler.amazonaws.com"
          },
          "Action": ["sts:AssumeRole"]
        }
      ]
    },
    "Policies": [
      {
        "PolicyName": "invoke",
        "PolicyDocument": {
          "Version": "2012-10-17",
          "Statement": [
            {
              "Effect": "Allow",
              "Action": ["lambda:InvokeFunction"],
              "Resource": /* lambdaLogicalId ARN */
            }
          ]
        }
      }
    ]
  }
}
```

...or you could create an unscoped version (`"Resource": "*"`) also.

If you're not open to this, could we at least update the documentation to note that this property is required when the method is set to `scheduler`? Thanks for your consideration.

Reply:
@drboyer I like this idea; I assume it'd be a good idea to check if a default role has been created (perhaps via a tag) and assign it rather than creating a new one for each function in a project? There are quotas for roles on an account, and as this PR is mainly meant to address an issue where devs can hit them, so I don't want to cause the same problem I'm trying to solve here.

Reply:
Instead of creating a new role behind the scenes, wouldn't it be ok, to rely on execution role that Framework is creating ?

Reply:
@medikoo Sounds good; we can extend that to include the `assumeRole` for `scheduler.amazonaws.com` and use it if one isn't passed. Will update.

Reply:
Might I add, if we go with the idea of creating a scheduler group for all schedules in a service, we may also want to do the same with the role?